contextual_orchestrator/model_discovery.py

```
@@ -0,0 +1,310 @@
+
+    def __init__(self, provider_name: str, detail: str) -> None:
+        self.provider_name = provider_name
+        super().__init__(f"model discovery failed for provider {provider_name!r}: {detail}")
+
+
+def _fetch_json(url: str, *, api_key: str, auth_scheme: str, timeout: float) -> Any:
+    request = urllib.request.Request(
+        url,
+        headers={"authorization": f"{auth_scheme} {api_key}"},
+        method="GET",
+    )
```

Ruff (0.16.1)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)